Fetch bundle uuids from the client in batches

kashizui opened the pull request:
Fixes #66.

@pbhatnagar3 please review :)

kashizui committed:
Fetch bundle uuids in batches from the client

kashizui committed:
Fix documentation in worksheet_util.get_bundle_uuid

percyliang commented on codalab/lib/worksheet_util.py:
```diff
@@ -167,24 +167,33 @@ def request_lines(worksheet_info, client):
+        else:
+            unresolved.append(bundle_spec)
+
+    return resolved + client.get_bundle_uuids(worksheet_uuid, unresolved)
```
Don't we need to return the uuids in the same order as the bundle_specs?
This code seems might permute things.

kashizui committed:
Fix ordering of get_bundle_uuids result

kashizui commented:
To address the ordering problem, simplified the logic to using the batch call on all of the provided bundle_specs if at least one of them need resolving (i.e. we will need to call the client anyway). Does this make sense?

percyliang commented:
Could do that, but I think it's not much harder to make a call on only the
unresolved names, build a dict from names to uuids, and then take another
pass over the array of bundle_specs.

 -Percy

On Sat, Sep 26, 2015 at 11:12 AM, Stephen Koo notifications@github.com
wrote:

> To address the ordering problem, simplified the logic to using the batch
> call on all of the provided bundle_specs if at least one of them need
> resolving (i.e. we will need to call the client anyway). Does this make
> sense?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/codalab/codalab-cli/pull/143#issuecomment-143477112.

kashizui committed:
Fetch only uuids for unresolved bundle specs

kashizui commented:
Got it-- does this implementation of the dict idea make sense?

Stephen

kashizui commented:
(still pending testing, please don't merge yet)

percyliang commented:
Structure looks good to me!

On Sat, Sep 26, 2015 at 2:29 PM, Stephen Koo notifications@github.com
wrote:

> (still pending testing, please don't merge yet)
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/codalab/codalab-cli/pull/143#issuecomment-143497618.

kashizui commented:
OK to merge!

this works as expected
https://gist.github.com/kashizui/f5f4d111845ca7fd3802

percyliang commented:
Great!  Could you distill this into a test and add it to test-cli.py?
Also, make sure you test passing in a combination of uuids and names (with
repeated specs).  It suffices just to do it for cl info -f uuid and check
the output.

On Sun, Sep 27, 2015 at 12:27 PM, Stephen Koo notifications@github.com
wrote:

> OK to merge!
> 
> this works as expected
> https://gist.github.com/kashizui/f5f4d111845ca7fd3802
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/codalab/codalab-cli/pull/143#issuecomment-143589002.

kashizui commented:
Will do—what do you mean by repeated specs?

percyliang commented:
cl info a a a 0x23... a 0x23...

On Sun, Sep 27, 2015 at 1:54 PM, Stephen Koo notifications@github.com
wrote:

> Will do—what do you mean by repeated specs?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/codalab/codalab-cli/pull/143#issuecomment-143594519.

kashizui committed:
Add test for batch cli calls

Also changed mentions of /etc/issues to /etc/group in test-cli.py
to adapt it to work on Mac OS X.

kashizui commented:
Added a test for the batch call, and also updated test-cli.py to work on Mac, and also to continue to the remaining tests even if one fails.

percyliang commented:
LGTM
